sdk/core/System.ClientModel/tests/Pipeline/ClientPipelineFunctionalTests.cs

```diff
@@ -138,5 +138,334 @@ public async Task NonBufferedResponseDisposedAfterMessageDisposed()
+    //{
+    //    var testDoneTcs = new CancellationTokenSource();
+    //    int i = 0;
+
+    //    ClientPipeline pipeline = ClientPipeline.Create();
+    //    TaskCompletionSource<object> tcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+    //    using TestServer testServer = new TestServer(
+    //        async context =>
+    //        {
+    //            Interlocked.Increment(ref i);
+    //            tcs.SetResult(null!);
+    //            await Task.Delay(Timeout.Infinite, testDoneTcs.Token);
+    //        });
+
+    //    var cts = new CancellationTokenSource();
+    //    using PipelineMessage message = pipeline.CreateMessage();
+    //    message.Request.Uri = testServer.Address;
+    //    ResponseBufferingPolicy.SetBufferResponse(message, false);
+
+    //    var task = Task.Run(() => pipeline.SendSyncOrAsync(message, IsAsync));
+
+    //    // Wait for server to receive a request
+    //    await tcs.Task;
+
+    //    cts.Cancel();
+
+    //    TaskCanceledException? exception = Assert.ThrowsAsync<TaskCanceledException>(async () => await task);
+    //    Assert.AreEqual("The operation was canceled.", exception!.Message);
+    //    Assert.AreEqual(1, i);
+
+    //    testDoneTcs.Cancel();
+    //}
+
+    [Test]
+    public async Task RetriesBufferedBodyTimeout()
+    {
+        var testDoneTcs = new CancellationTokenSource();
+        int i = 0;
+        ClientPipelineOptions options = new() { NetworkTimeout = TimeSpan.FromMilliseconds(500) };
+        ClientPipeline pipeline = ClientPipeline.Create(options);
+
+        using TestServer testServer = new TestServer(
+            async context =>
+            {
+                if (Interlocked.Increment(ref i) == 1)
+                {
+                    context.Response.StatusCode = 200;
+                    context.Response.Headers.ContentLength = 10;
+                    await context.Response.WriteAsync("1");
+                    await context.Response.Body.FlushAsync();
+
+                    await Task.Delay(Timeout.Infinite, testDoneTcs.Token);
+                }
+                else
+                {
+                    context.Response.StatusCode = 201;
+                    await context.Response.WriteAsync("Hello world!");
+                }
+            });
+
+        using PipelineMessage message = pipeline.CreateMessage();
+        message.Request.Uri = testServer.Address;
+        message.BufferResponse = true;
+
+        await pipeline.SendSyncOrAsync(message, IsAsync);
+
+        Assert.AreEqual(message.Response!.Status, 201);
+        Assert.AreEqual("Hello world!", await new StreamReader(message.Response.ContentStream!).ReadToEndAsync());
+        Assert.AreEqual("Hello world!", message.Response.Content.ToString());
+        Assert.AreEqual(2, i);
+
+        testDoneTcs.Cancel();
+    }
+
+    #endregion
+
+    #region Test parallel connections
+
+    // TODO: This one hangs on net462.  Why?
```

Comment:
TODO: Fix this test

Reply:
Tracking here: https://github.com/Azure/azure-sdk-for-net/pull/41325#discussion_r1456628977